android/gltfio-android/src/main/cpp/AssetLoader.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <jni.h>
+
+#include <filament/Engine.h>
+
+#include <utils/EntityManager.h>
+
+#include <gltfio/AssetLoader.h>
+#include <gltfio/MaterialProvider.h>
+
+#include "common/NioUtils.h"
+
+using namespace filament;
+using namespace gltfio;
+using namespace utils;
+
+extern void registerCallbackUtils(JNIEnv*);
+extern void registerNioUtils(JNIEnv*);
+
+jint JNI_OnLoad(JavaVM* vm, void* reserved) {
+    JNIEnv* env;
+    if (vm->GetEnv(reinterpret_cast<void**>(&env), JNI_VERSION_1_6) != JNI_OK) {
+        return -1;
+    }
+
+    registerCallbackUtils(env);
```

Comment:
Does this work? Since you include the filament-android build I would expect the utils to be registered twice.

Reply:
Yes in fact it doesn't work without it. Filament is being loaded dynamically in my upcoming demo, as evidenced by the EventManager singleton bug that I ran into.